pkg/kubelet/util/util_unix_test.go

```diff
@@ -133,3 +133,41 @@ func TestIsUnixDomainSocket(t *testing.T) {
 		assert.Equal(t, result, test.expectSocket, "Unexpected result from IsUnixDomainSocket: %v for %s", result, test.label)
 	}
 }
+
+func TestGetAddressAndDialer(t *testing.T) {
+	tests := []struct {
+		endpoint     string
+		expectError  bool
+		expectedAddr string
+	}{
+		{
+			endpoint:     "unix:///tmp/s1.sock",
+			expectError:  false,
+			expectedAddr: "/tmp/s1.sock",
+		},
+		{
+			endpoint:     "unix:///tmp/f6.sock",
+			expectError:  false,
+			expectedAddr: "/tmp/f6.sock",
+		},
+		{
+			endpoint:    "tco:/localhost:9090",
+			expectError: true,
+		},
+		{
+			endpoint:    "htta://free-test.com",
```

Comment:
should this be "http://..."?

Reply:
I think the misspelling is actually intentional, so that it fails :) But perhaps a comment to that effect both for this test and the one above could be helpful :)

Reply:
> I think the misspelling is actually intentional, so that it fails :) But perhaps a comment to that effect both for this test and the one above could be helpful :)

yeah, it's  intentional, I will add useful comment for it. 😀